src/root/static/js/common.js

```diff
@@ -46,29 +46,17 @@ $(document).ready(function() {
           window.location = link.attr("href");
     });
 
-    bootbox.animate(false);
+    bootbox.setDefaults({ animate: false });
 
     $(".hydra-popover").popover({});
 
-    $(function() {
-        if (window.location.hash) {
-            $(".nav-tabs a[href='" + window.location.hash + "']").tab('show');
-        }
-
-        /* If no tab is active, show the first one. */
-        $(".nav-tabs").each(function() {
-            if ($("li.active", this).length > 0) return;
-            $("a", $(this).children("li:not(.dropdown)").first()).tab('show');
-        });
+    /* Deep emedding for tabs. */
+    if (window.location.hash) {
+        $('.nav-tabs > .nav-item:not(.dropdown) a[href="' + window.location.hash + '"]').tab('show');
+    }
 
-        /* Ensure that pressing the back button on another page
-           navigates back to the previously selected tab on this
-           page. */
-        $('.nav-tabs').bind('show', function(e) {
-            var pattern = /#.+/gi;
-            var id = e.target.toString().match(pattern)[0];
-            history.replaceState(null, "", id);
-        });
+    $('.nav-tabs > .nav-item:not(.dropdown) a[href^="#"]').on('click', function() {
+        history.pushState(null, null, window.location.href.split("#")[0] + $(this).attr("href"));
```

Comment:
I see why `replaceState` must have been used. Easier to deal with than `pushState` when you have "simple" JavaScript like that.

On a page, going to a tab, then using the back navigation does not go to the tab the URL is now showing (unless you reload). It looks like it's "stealing" a step from going back, since it does nothing.
Since the current Hydra uses `replaceState`, let's use it, otherwise I'm betting we'll have a deluge of "now Hydra is eating back button" issues like I've seen happen on projects switching to the (in my opinion) more correct pushState. (The other options would have been to add the listener to history events.)

```suggestion
        history.replaceState(null, null, window.location.href.split("#")[0] + $(this).attr("href"));
```

Reply:
Thanks a ton for all this feedback! :+1: I'll mark these as resolved as I knock them off.